### ISSUE TITLE
Game crash when trying to change shulker boxes

### ISSUE THREAD

**Issue by yannicksuc** [Digifox03/variedMobs#12]
I've just made a basic ressource pack for changing the texture of shulker boxes.
So in varied\textures\entity\shulker\shulker.json
`{
  "type": "varied-mobs:name",
  "regex": "miam",
  "value": {
  	"type": "varied-mobs:result",
  	"result": "textures/entity/shulker/shulker1.png"
	}
}`

And the game crash when I have a shulker named "miam" in my world...

Also, I can't find a way to change textures of colored shulkers, I've tried with a json named yellow_shulker for example and it's not working

Here is the crash logs :

`---- Minecraft Crash Report ----
// Don't be sad, have a hug! <3

Time: 20/01/21 00:19
Description: Rendering entity in world

java.lang.NullPointerException: Rendering entity in world
	at it.digifox03.variedmobs.selectors.NameSelector.prop(NameSelector.java:132)
	at it.digifox03.variedmobs.selectors.BoolSelector.choose(BoolSelector.java:118)
	at it.digifox03.variedmobs.core.VariedMobManager.runRedirect(VariedMobManager.java:36)
	at it.digifox03.variedmobs.core.VariedMobManager.redirectTexture(VariedMobManager.java:24)
	at net.minecraft.class_922.localvar$zzl000$getRenderLayer(class_922.java:538)
	at net.minecraft.class_922.method_24302(class_922.java:226)
	at net.minecraft.class_922.method_4054(class_922.java:178)
	at net.minecraft.class_927.method_4072(class_927.java:47)
	at net.minecraft.class_927.method_3936(class_927.java:20)
	at net.minecraft.class_898.method_3954(class_898.java:310)
	at net.minecraft.class_761.method_22977(class_761.java:2088)
	at net.minecraft.class_761.method_22710(class_761.java:1724)
	at net.minecraft.class_757.method_3188(class_757.java:1022)
	at net.minecraft.class_757.method_3192(class_757.java:693)
	at net.minecraft.class_310.method_1523(class_310.java:1048)
	at net.minecraft.class_310.method_1514(class_310.java:681)
	at net.minecraft.client.main.Main.main(Main.java:215)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:224)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:141)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:27)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at it.digifox03.variedmobs.selectors.NameSelector.prop(NameSelector.java:132)
	at it.digifox03.variedmobs.selectors.BoolSelector.choose(BoolSelector.java:118)
	at it.digifox03.variedmobs.core.VariedMobManager.runRedirect(VariedMobManager.java:36)
	at it.digifox03.variedmobs.core.VariedMobManager.redirectTexture(VariedMobManager.java:24)
	at net.minecraft.class_922.localvar$zzl000$getRenderLayer(class_922.java:538)
	at net.minecraft.class_922.method_24302(class_922.java:226)
	at net.minecraft.class_922.method_4054(class_922.java:178)
	at net.minecraft.class_927.method_4072(class_927.java:47)
	at net.minecraft.class_927.method_3936(class_927.java:20)

-- Entity being rendered --
Details:
	Entity Type: minecraft:shulker (net.minecraft.class_1606)
	Entity ID: 34
	Entity Name: miam
	Entity's Exact location: 142.50, 56.00, 114.50
	Entity's Block location: World: (142,56,114), Chunk: (at 14,3,2 in 8,7; contains blocks 128,0,112 to 143,255,127), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Entity's Momentum: 0.00, 0.00, 0.00
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null

-- Renderer details --
Details:
	Assigned renderer: net.minecraft.class_943@3ca38e3d
	Location: -1.40,-2.24,-2.59 - World: (-2,-3,-3), Chunk: (at 14,-1,13 in -1,-1; contains blocks -16,0,-16 to -1,255,-1), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
	Rotation: -68.90625
	Delta: 0.4001149
Stacktrace:
	at net.minecraft.class_898.method_3954(class_898.java:310)
	at net.minecraft.class_761.method_22977(class_761.java:2088)
	at net.minecraft.class_761.method_22710(class_761.java:1724)
	at net.minecraft.class_757.method_3188(class_757.java:1022)

-- Affected level --
Details:
	All players: 1 total; [class_746['LordFinn'/137, l='ClientLevel', x=143.90, y=56.62, z=117.09]]
	Chunk stats: Client Chunk Cache: 225, 108
	Level dimension: minecraft:overworld
	Level spawn location: World: (8,55,8), Chunk: (at 8,3,8 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 749994 game time, 13000 day time
	Server brand: fabric
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:617)
	at net.minecraft.class_310.method_1587(class_310.java:2239)
	at net.minecraft.class_310.method_1514(class_310.java:699)
	at net.minecraft.client.main.Main.main(Main.java:215)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:224)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:141)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:27)

-- System Details --
Details:
	Minecraft Version: 1.16.4
	Minecraft Version ID: 1.16.4
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 1079696128 bytes (1029 MB) / 2080374784 bytes (1984 MB) up to 3221225472 bytes (3072 MB)
	CPUs: 8
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx3G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		appleskin: AppleSkin 1.0.11
		autoconfig1u: Auto Config v1 Updated 3.3.1
		cloth-basic-math: Cloth Basic Math 0.5.1
		cloth-client-events-v0: Cloth Client Events v0 1.4.5
		cloth-config2: Cloth Config v4 4.8.3
		fabric: Fabric API 0.28.3+1.16
		fabric-api-base: Fabric API Base 0.2.0+ab87788d3a
		fabric-biome-api-v1: Fabric Biome API (v1) 3.1.0+2e23b97c3a
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.4+6a2618f53a
		fabric-command-api-v1: Fabric Command API (v1) 1.0.9+6a2618f53a
		fabric-commands-v0: Fabric Commands (v0) 0.2.0+6a2618f53a
		fabric-containers-v0: Fabric Containers (v0) 0.1.9+a03e98793a
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.2.0+e77439c73a
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.2+b7f9825d3a
		fabric-dimensions-v1: fabric-dimensions-v1 2.0.1+9a6c75813a
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.1+6a2618f53a
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.0+6a2618f53a
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.3+a4467d2a3a
		fabric-item-api-v1: Fabric Item API (v1) 1.2.0+6a2618f53a
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.2.1+6a2618f53a
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.1+730711c63a
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.0+6a2618f53a
		fabric-language-kotlin: Fabric Language Kotlin 1.4.21+build.1
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.2.0+74cc3b2095
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.1+6a2618f53a
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.2+6a2618f53a
		fabric-models-v0: Fabric Models (v0) 0.1.1+6a2618f53a
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.0+4358fbc63a
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.7+a03e98793a
		fabric-networking-v0: Fabric Networking (v0) 0.3.1+2a4333d33a
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.9.2+6a2618f53a
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.1+6a2618f53a
		fabric-particles-v1: fabric-particles-v1 0.2.2+6a2618f53a
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.3+be155ae23a
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.3.3+6a2618f53a
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.3+6a2618f53a
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 2.2.0+6a2618f53a
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.4+6a2618f53a
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.12+6a2618f53a
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.1+6a2618f53a
		fabric-rendering-v1: Fabric Rendering (v1) 1.4.0+6a2618f53a
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.3.2+1944f3e03a
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.0+6a2618f53a
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.1+f1d8af063a
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.1.0+e77439c73a
		fabric-textures-v0: Fabric Textures (v0) 1.0.5+6a2618f53a
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.5+6a2618f53a
		fabricloader: Fabric Loader 0.10.8
		fiber: fiber 0.23.0-2
		inventoryprofiles: Inventory Profiles 0.4.2
		litematica: Litematica 0.0.0-dev.20201103.184101
		malilib: MaLiLib 0.10.0-dev.21+arne.2
		minecraft: Minecraft 1.16.4
		mm: Manningham Mills 2.1
		okzoomer: Ok Zoomer 4.0.1+1.16.2
		optifabric: OptiFabric 1.8.13
		org_jetbrains_annotations: annotations 20.0.0
		org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.4.21
		org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.4.21
		org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.4.21
		org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.4.21
		org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.4.2
		org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.4.2
		roughlyenoughitems: Roughly Enough Items 5.8.10
		roughlyenoughitems-api: REI (API) 5.8.10
		roughlyenoughitems-default-plugin: REI (Default Plugin) 5.8.10
		roughlyenoughitems-runtime: REI (Runtime) 5.8.10
		shulkerboxtooltip: Shulker Box Tooltip 2.3.3+1.16.4
		varied-mobs: Varied Mobs 1.0.0
		worldedit: WorldEdit 7.2.1+299386a
	Launched Version: fabric-loader-0.10.8-1.16.4
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: Intel(R) UHD Graphics 620 GL version 4.6.0 - Build 27.20.100.8190, Intel
	GL Caps: Using framebuffer using OpenGL 3.0
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Graphics mode: fast
	Resource Packs: vanilla, fabric/appleskin (incompatible), fabric/autoconfig1u, fabric/cloth-config2, fabric/fabric, fabric/fabric-api-base, fabric/fabric-biome-api-v1, fabric/fabric-blockrenderlayer-v1, fabric/fabric-command-api-v1, fabric/fabric-containers-v0, fabric/fabric-content-registries-v0, fabric/fabric-crash-report-info-v1, fabric/fabric-events-interaction-v0, fabric/fabric-events-lifecycle-v0, fabric/fabric-game-rule-api-v1, fabric/fabric-item-api-v1, fabric/fabric-item-groups-v0, fabric/fabric-key-binding-api-v1, fabric/fabric-keybindings-v0, fabric/fabric-lifecycle-events-v1, fabric/fabric-loot-tables-v1, fabric/fabric-mining-levels-v0, fabric/fabric-models-v0, fabric/fabric-networking-api-v1, fabric/fabric-networking-blockentity-v0, fabric/fabric-networking-v0, fabric/fabric-object-builder-api-v1, fabric/fabric-object-builders-v0, fabric/fabric-registry-sync-v0, fabric/fabric-renderer-api-v1, fabric/fabric-renderer-indigo, fabric/fabric-renderer-registries-v1, fabric/fabric-rendering-data-attachment-v1, fabric/fabric-rendering-fluids-v1, fabric/fabric-rendering-v0, fabric/fabric-rendering-v1, fabric/fabric-resource-loader-v0, fabric/fabric-screen-handler-api-v1, fabric/fabric-tag-extensions-v0, fabric/fabric-textures-v0, fabric/fabric-tool-attribute-api-v1, fabric/fabricloader, fabric/inventoryprofiles, fabric/litematica (incompatible), fabric/malilib (incompatible), fabric/mm, fabric/okzoomer, fabric/optifabric, fabric/roughlyenoughitems, fabric/roughlyenoughitems-runtime, fabric/shulkerboxtooltip, fabric/worldedit (incompatible), file/example-pack.zip (incompatible), fabric/fabric-language-kotlin, fabric/varied-mobs, file/two - Copy
	Current Language: English (US)
	CPU: 8x Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz
	OptiFine Version: OptiFine_1.16.4_HD_U_G5
	OptiFine Build: 20201106-171901
	Render Distance Chunks: 5
	Mipmaps: 1
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 4.6.0 - Build 27.20.100.8190
	OpenGlRenderer: Intel(R) UHD Graphics 620
	OpenGlVendor: Intel
	CpuCount: 8

-- OptiFabric --
Details:
	OptiFine jar designed for: 1.16.4
	OptiFine jar version: OptiFine_1.16.4_HD_U_G5
	OptiFine jar status: Valid OptiFine installer
	OptiFine remapped jar: C:\Users\Yannick\AppData\Roaming\.minecraft\.optifine\OptiFine_1.16.4_HD_U_G5\Optifine-mapped.jar
	OptiFabric error: <None>`

**Comment by Digifox03** [comment]
Please try reproducing the error without using other mods.
Also, make sure to insert the log inside a code block

**Comment by supsm** [comment]
I think this is an issue with `varied-mobs:name`. I'm getting crashes when I include it, but not when I don't. Same as above, I do not get crashes when there is no matching name.

Here is a snippet of the json
```
{
	"regex": "name",
	"type": "varied-mobs:name",
	"value": {
		"choices": [
			"minecraft:varied/textures/entity/zombie/zombie21.png"
			"minecraft:varied/textures/entity/zombie/zombie22.png"
		],
		"type": "varied-mobs:pick"
	}
}
```

Unfortunately my logs are very long (several hundred MB) so I can't post them here

**Comment by supsm** [comment]
I've done some further investigation, the issue is that `reg` in `NameSelector` is always `null`. I've tried everything I could to fix this, but nothing works

**Comment by Digifox03** [comment]
solved by 29f3d99c8f1509115e6d1aeb4e11c6915d819467 (coming with 1.17 update)
